fieldsolver/gridGlue.cpp

```diff
@@ -94,31 +94,27 @@ void filterMoments(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
          for (FsGridTools::FsIndex_t j = 0; j < mntDims[1]; j++){
             for (FsGridTools::FsIndex_t i = 0; i < mntDims[0]; i++){
 
-               //  Get refLevel level
                int refLevel = technicalGrid.get(i, j, k)->refLevel;
+               auto* swap {swapGrid.get(i, j, k)};
 
-               // Skip pass
+               // Skip pass and copy value
                if (blurPass >= P::numPasses.at(refLevel) || technicalGrid.get(i, j, k)->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) {
+                  *swap = *momentsGrid.get(i, j, k);
                   continue;
                }
 
-               // Get pointers to our cells
-               std::array<Real, fsgrids::moments::N_MOMENTS> *cell;  
-               std::array<Real, fsgrids::moments::N_MOMENTS> *swap;
-            
                // Set Cell to zero before passing filter
-               swap = swapGrid.get(i,j,k);
-               for (int e = 0; e < fsgrids::moments::N_MOMENTS; ++e) {
-                  swap->at(e)=0.0;
+               for (auto& moment : *swap) {
```

Comment:
I completely disagree about it being worse, sure that could be `real&` but the range-based loop clarifies what's actually going on rather than obfuscating it. I'll begrudgingly accept that it's bad that the two loops look different now though and revert that one, since we don't have C++23 and `zip` yet.

It also like I also managed to faff up the scope on `cell` which is thankfully not too important.

Reply:
My opinion:

Semantically, the contents of a fsgrid container really aren't "something to loop over". Even though they are stored as arrays of floating point numbers, they really are just structs in a different dress (mostly for historical reasons).
So while "go through all these numbers and set them to zero" is something that a range-based loop does correctly, it does not correspond to the semantics of what's going on here.

If you want to optimize it, you can use a memset call here. :)

Reply:
I did actually consider `*swap = std::array<Real, N_MOMENTS> {0}`. Not sure if that would end up optimizing to `memset`, but it probably doesn't matter too much either way 😄.